crates/ty_python_semantic/src/types.rs

```diff
@@ -1655,6 +1655,20 @@ impl<'db> Type<'db> {
                 }
             }
 
+            (
+                Type::StringLiteral(_)
+                | Type::LiteralString
+                | Type::BooleanLiteral(_)
+                | Type::IntLiteral(_)
+                | Type::BytesLiteral(_)
+                | Type::ModuleLiteral(_),
+                _,
+            ) if (self.literal_fallback_instance(db))
+                .is_some_and(|instance| instance.is_assignable_to(db, target)) =>
+            {
+                true
+            }
+
             (Type::ClassLiteral(class_literal), Type::Callable(_)) => {
```

Comment:
I think we also need to add the same logic for `LiteralString`:

```suggestion
            (Type::LiteralString, _) if KnownClass::Str.to_instance(db).is_assignable_to(db, target) => true,

            (Type::ClassLiteral(class_literal), Type::Callable(_)) => {
```

Reply:
(need to add a test for this as well)

Reply:
i added a test for it

Reply:
oh i see how this is different sorry